### PR TITLE
Add spec for Atan2Op

### LINKED ISSUE #465: Add spec for Atan2Op

### PR DESCRIPTION
**Pull request by ghpvnist**
closes #465

### CODE REVIEW

**Comment by ghpvnist** [comment]
Verification is set to `revisit` since it is missing verification check to allow only floating-point and complex element types. Tracking this issue in #499.

File: docs/spec_draft.md [hunk 1]
```diff
@@ -334,6 +335,46 @@ For boolean tensors, computes the logical operation.
...
 [Back to Ops](#index-of-ops)
 
+## stablehlo.atan2
+
+### Semantics
+
+Performs element-wise atan2 operation on `lhs` and `rhs` tensor and produces a
+`result` tensor, implementing the `atan2` operation from the IEEE-754
+specification. For complex element types, it computes a complex atan2 function
+below, with corner cases TBD. Numeric precision is implementation-defined.
```
Comment (sdasgup3) [inline docs/spec_draft.md:345]: defined below

File: docs/spec_draft.md [hunk 2]
```diff
@@ -334,6 +335,46 @@ For boolean tensors, computes the logical operation.
...
+`log` and `sqrt` operations correspond to [stablehlo.log](#stablehlolog) and
+[stablehlo.sqrt](#stablehlosqrt):
+$$atan2(lhs, rhs) = -i * \log\left(\frac{rhs + i * lhs}{sqrt(rhs^{2} + lhs^{2})}\right)$$
```
[the remark this answers is not in the document]
Comment (sdasgup3) [reply docs/spec_draft.md:348]: Can we make sure that the y and x mentioned at https://github.com/tensorflow/tensorflow/blob/463d656802b67b377266b68d4b75db6bf8462f2b/tensorflow/compiler/xla/hlo/evaluator/hlo_evaluator_typed_visitor.h#L546 are the real and imag part resp?

[the remark this answers is not in the document]
Comment (ghpvnist) [reply docs/spec_draft.md:348]: atan2(y, x) is the widely known format where y is the imaginary and x is the real: https://en.wikipedia.org/wiki/Atan2
The calculation shown here is also aligned with what wolfram alpha and other widely known frameworks use (including HLO).

[the remark this answers is not in the document]
Comment (sdasgup3) [reply docs/spec_draft.md]: Thanks!
I missed the part that `lhs` and `rhs` are complex values themselves.

**Review by sdasgup3 — APPROVED** [review]

File: docs/spec_draft.md [hunk 3]
```diff
@@ -336,6 +337,47 @@ For boolean tensors, computes the logical operation.
...
+| Name  | Type                                     |
+|-------|------------------------------------------|
+| `lhs` | tensor of floating-point or complex type |
+| `rhs` | tensor of floating-point or complex type |
+
+### Outputs
+
+| Name     | Type                                     |
+|----------|------------------------------------------|
+| `result` | tensor of floating-point or complex type |
```
Comment (burmako) [inline docs/spec_draft.md:364]: [shape_inference.cc](https://github.com/tensorflow/tensorflow/blob/2c1b3efe573befdda3f3db399eebb875e690082c/tensorflow/compiler/xla/service/shape_inference.cc#L1028-L1043) does allow integers, although [elemental_ir_emitter.cc](https://github.com/tensorflow/tensorflow/blob/2c1b3efe573befdda3f3db399eebb875e690082c/tensorflow/compiler/xla/service/elemental_ir_emitter.cc#L1868-L1870) does not. Which piece of infrastructure is right about this?

File: docs/spec_draft.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (burmako) [reply docs/spec_draft.md]: Since the formula is well-known, do we have to provide it in our spec?

File: docs/spec_draft.md (hunk 3 above)
Comment (ghpvnist) [reply docs/spec_draft.md:364]: Most math operations do not allow integers, so `shape_inference.cc` is likely in the wrong here. `Tanh`, which is the most similar op to this one, though a unary op, does not allow integers. Should I file an issue and set Type Inference to `revisit`?

File: docs/spec_draft.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ghpvnist) [reply docs/spec_draft.md]: Since we don't have formulas written for other similar specs, I'll remove them.

File: docs/spec_draft.md (hunk 3 above)
Comment (burmako) [reply docs/spec_draft.md:364]: Sounds good to keep the status quo, i.e. set Verification (not Type Inference) to "revisit" and keep #499 open. No further comments on this PR - LGTM.

**Review by burmako — APPROVED** [review]

**Pull request merged**